core/app/models/spree/shipping_rate_estimator.rb

```
@@ -55,21 +58,23 @@ def package
 
     # The shipping rates for order
     #
-    # @return [Enumerable<Spree::ShippingRate>]
+    # @return [Enumerable<Spree::shippingRateEstimator::Estimation>]
     def shipping_rates
       shipping_methods.map do |shipping_method|
-        shipping_method.shipping_rates.new(cost: calculate(shipping_method))
+        Estimation.new(
+          shipping_method: shipping_method,
+          cost:            calculate(shipping_method)
+        )
       end
     end
 
     # Calculate a rate for shipping method
     #
     # @param [Spree::ShippingMethod] shipping_method
     #
-    # @return [Float]
-    #   yes float is sadly the return type currently, blame spree_active_shipping
+    # @return [BigDecimal]
     def calculate(shipping_method)
-      shipping_method.calculator.compute(package)
+      shipping_method.calculator.compute(package).to_d # Yes spree_active_shipping returns a float
```

I've to reject this as the explicit coercion is only needed when the silently coercing `Spree::ShippingRate` model is gone.